addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JSFunctionPassiveScanner.java

```diff
@@ -0,0 +1,203 @@
+            }
+        } else if (msg.getResponseHeader().isJavaScript()) {
+            // Raw search on response body
+            String content = msg.getResponseBody().toString();
+            for (Pattern pattern : patterns) {
+                if (pattern.matcher(content).find()) {
+                    evidence.append(pattern);
+                    break; // Only need to record one instance of vulnerability
+                }
+            }
+        }
+        if (evidence.length() > 0) {
+            this.raiseAlert(msg, id, evidence.toString());
+        }
+    }
+
+    private void raiseAlert(HttpMessage msg, int id, String evidence) {
+        Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_LOW, getName());
+        alert.setDetail(
+                this.getDescription(),
+                msg.getRequestHeader().getURI().toString(),
+                "", // Param, not relevant for this vulnerability
+                "", // Attack, not relevant for passive vulnerabilities
+                "", // Other info not specified in message bundle
+                this.getSolution(),
+                this.getReference(),
+                evidence,
+                749, // CWE-749: Exposed Dangerous Method or Function
+                0, // WASC Id - return 0 if no relevant one
+                msg);
+
+        parent.raiseAlert(id, alert);
+    }
+
+    private static void createPatterns() {
+        patterns = new HashSet<>();
+        defaultPatterns = new HashSet<>();
+        try {
+            File f =
+                    new File(
+                            Constant.getZapHome()
+                                    + File.separator
+                                    + FUNC_LIST_DIR
+                                    + File.separator
+                                    + FUNC_LIST_FILE);
+            if (!f.exists()) {
+                throw new IOException("Couldn't find resource: " + f.getAbsolutePath());
+            }
+            try (BufferedReader reader = new BufferedReader(new FileReader(f))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    line = line.trim();
+                    if (!line.startsWith("#") && line.length() > 0) {
+                        defaultPatterns.add(
+                                Pattern.compile("\\b" + line + "\\b", Pattern.CASE_INSENSITIVE));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error(
+                    "Error on opening/reading js functions file: "
+                            + File.separator
+                            + FUNC_LIST_DIR
+                            + File.separator
+                            + FUNC_LIST_FILE
+                            + " Error: "
+                            + e.getMessage());
+        }
+    }
+
+    private static void loadPayload() {
+        patterns = defaultPatterns;
```

Comment:
Was this meant to be `patterns = new HashSet(defaultPatterns);`? Otherwise it will change the `defaultPatterns`.

Reply:
Yes it was sorry, thanks for catching that!